api/python/provisioner/commands/check.py

```diff
@@ -0,0 +1,578 @@
+            #  check cross connectivity if nodes > 2 and all possible pairs
+            targets = set(servers) - {addr}
+            targets = (cfg.LOCAL_MINION if not targets
+                       else next(iter(targets)))  # takes just one node
+
+            cmd = _PING_CMD_TMPL.format(server_addr=addr)
+            try:
+                cmd_run(cmd, targets=targets)
+            except SaltCmdResultError:
+                res.set_fail(target=targets,
+                             comment=(f"{cfg.CheckVerdict.FAIL.value}: {addr} "
+                                      f"is not reachable from {targets}"))
+            else:
+                res.set_passed(target=targets)
+
+        return res
+
+    @staticmethod
+    def _logs_are_good(*, args: str) -> List[CheckEntry]:
+        """
+        Check that logs are clear and don't contain some specific key phrases
+        which signal about serious issues
+
+        :param args: logs_are_good check specific parameters and arguments
+        :return:
+        """
+        # TODO: do we need to fail if logfile doesn't exist?
+        _LOGS_ARE_GOOD_CHECK_CMD_TMPL = (
+                                    'test -f {log_filename} || '
+                                    '{{ echo "logfile does not exist" && '
+                                    'exit 1 ; }} ; '
+                                    'grep -i "{key_phrase}" {log_filename} '
+                                    '1>/dev/null && '
+                                    'echo "{key_phrase} exists in logfile" && '
+                                    'exit 1')
+
+        minion_id = local_minion_id()
+        res: List[CheckEntry] = list()
+
+        def check_log_file(log_file, key_phrases, _targets):
+            """
+            Check separate log_file against key_phrases
+
+            :param log_file:
+            :param key_phrases:
+            :param _targets
+            :return:
+            """
+            _res: CheckEntry = CheckEntry(cfg.Checks.LOGS_ARE_GOOD.value)
+            for phrase in key_phrases:
+                cmd = _LOGS_ARE_GOOD_CHECK_CMD_TMPL.format(
+                    log_filename=log_file,
+                    key_phrase=phrase)
+                try:
+                    cmd_run(cmd, targets=_targets)
+                except SaltCmdResultError as e:
+                    _res.set_fail(target=minion_id,
+                                  comment=f"{log_file}: {str(e.reason)}")
+                    return _res
+
+            _res.set_passed(target=minion_id, comment=f"{log_file}")
+
+            return _res
+
+        # grep "unable to stop resource" /var/log/pacemaker.log
+        # grep "reboot" /var/log/corosync.log
+        # grep "error" /var/log/corosync.log"
+
+        # they should be a constants
+        corosync_log = "/var/log/cluster/corosync.log"
+        pacemaker_log = "/var/log/pacemaker.log"
+
+        corosync_key_phrases = (
+            "reboot",
```

Comment:
I think these may lead to false positives since:
- logs may include outdated records (e.g. an issue has been fixed already and cluster is good now)
- some words might not be directly related to a bad state of the cluster since it is very specific to how different components treat different cases (e.g. 'reboot might be required') or depends on context (e.g. '0 errors were detected')

@azheregelya Could you provide your vision on that: is it ok at all to rely on any phrases in logs to make a desicion regarding the whole system state? Do you think there are some good phrares for that?

@dmitrydb Generally I think it would be safer to exclude that check from SWUpdate routine:
- either skip it totally
- or do not fail if any phrases are found and just warn

Reply:
Interesting phrases:
`Forcing <resource-name> away from <node-name>`
`node is unclean`
`can fence`
`node <node-name> will be fenced`

Existing example with "unable to stop resource" is good as well.
Recommended file /var/log/cluster/corosync.log

Reply:
Another important moment - we need make sure that only new events are considered. This becomes relevant in case of update when system already has some "history" in corosync.log

Reply:
I guess to move these improvements in separate PR to check and debug that logic separately from the main flow.